fs/inode/fs_inodesearch.c

```diff
@@ -467,7 +490,20 @@ int inode_search(FAR struct inode_search_s *desc)
    * node if node is a symbolic link.
    */
 
-  DEBUGASSERT(desc != NULL);
+  DEBUGASSERT(desc != NULL && desc->path != NULL);
+
+  /* Convert the relative path to the absolute path */
+
+  if (*desc->path != '/')
+    {
+      asprintf(&desc->buffer, "%s/%s", _inode_getcwd(), desc->path);
+      if (desc->buffer == NULL)
```

Comment:
Is there a memory leak here?  Where is desc->buffer freed?

Reply:
Yes, it's release in RELEASE_SEARCH. I reuse the same code which release the soft link buffer.